File: esthesis-server/esthesis-server-impl/src/main/java/esthesis/platform/backend/server/dto/DashboardDTO.java
package esthesis.platform.backend.server.dto;

import esthesis.platform.backend.common.dto.BaseDTO;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class DashboardDTO extends BaseDTO {

  private String userId;
  private boolean shared;
  private String name;
  private boolean defaultView;
}

File: esthesis-core-ui/src/app/dashboard/dto/view-edit/dashboard-dto.ts
import {BaseDto} from "../../../dto/base-dto";
import {DashboardItemDto} from "./dashboard-item-dto";

export interface DashboardDto extends BaseDto {
  id: string;
  name: string;
  description: string;
  shared: boolean;
  home: boolean;
  items: DashboardItemDto[];
}
